sentry_sdk/integrations/tornado.py

```diff
@@ -110,6 +91,39 @@ def sentry_log_exception(self, ty, value, tb, *args, **kwargs):
         RequestHandler.log_exception = sentry_log_exception  # type: ignore
 
 
+@contextlib.contextmanager
+def _handle_request_impl(self):
+    # type: (RequestHandler) -> Generator[None, None, None]
+    hub = Hub.current
+    integration = hub.get_integration(TornadoIntegration)
+
+    if integration is None:
+        yield
+
+    weak_handler = weakref.ref(self)
+
+    with Hub(hub) as hub:
+        with hub.configure_scope() as scope:
+            scope.clear_breadcrumbs()
+            processor = _make_event_processor(weak_handler)  # type: ignore
```

Comment:
Does this not have the same issue as #1034 , the `weak_handler` is not accessible within the `tornado_processor` https://github.com/getsentry/sentry-python/blob/master/sentry_sdk/integrations/tornado.py#L135 ?

Reply:
Good catch, but I believe `self` will stick around for the duration of the request handler invocation, which sticks around for the duration of the response streaming, so it should be fine afaict

let me add a test

Reply:
I don't think the test you added covers that case since that request raises an exception and so triggers the error handler.
I would create another test to check if the request body exists when the request doesn't raise an exception and returns a success status code.

Reply:
Conceptually the exception won't make the request stay around for longer, but you're right it's not the same testcase so I added another one.